src/shared/machine-pool.c

```diff
@@ -31,8 +31,14 @@ int setup_machine_directory(sd_bus_error *error) {
         if (r == 0)
                 return 0;
 
+        if (!use_btrfs_subvol)
+                return 1;
+
         (void) btrfs_subvol_make_label("/var/lib/machines");
 
+        if (!use_btrfs_quota)
+                return 1;
+
         r = btrfs_quota_enable("/var/lib/machines", true);
         if (r < 0)
                 log_warning_errno(r, "Failed to enable quota for /var/lib/machines, ignoring: %m");
```

Comment:
This is a pre-existing condition, but please fix it when touching this and don't duplicate it in your patch.
The callers of this function do not care whether 0 or 1 is returned, and there's no clear meaning to the return value. Thus please make your additions return 0, and then add a second commit which changes the existing `return 1` to `return 0`.

Reply:
Makes sense. I was not sure how far should I go rearranging pre-existing conventions in a minor PR like this :)